### PR TITLE
PUBDEV-6046: Failed to get metric: auc from ModelMetrics type BinomialGLM

### PR DESCRIPTION
**Pull request by michalkurka**
This PR makes sure the AUC2 object is created even in edge cases when the data have 0 total weight. This prevents possible NPEs.

### CODE REVIEW

File: h2o-core/src/main/java/hex/ModelMetricsBinomial.java [hunk 1]
```diff
@@ -210,12 +210,14 @@ private ModelMetrics makeModelMetrics(Model m, Frame f, GainsLift gl) {
       double mse = Double.NaN;
       double logloss = Double.NaN;
       double sigma = Double.NaN;
-      AUC2 auc = null;
+      final AUC2 auc;
```
Comment (pkozelka) [inline h2o-core/src/main/java/hex/ModelMetricsBinomial.java]: cool, I like immutable vars wherever possible ;)

File: h2o-core/src/main/java/hex/ModelMetrics.java [hunk 2]
```diff
@@ -150,16 +156,12 @@ static public double getMetricFromModelMetric(ModelMetrics mm, String criterion)
...
-    } catch(Exception fallthru) {
-      try {
-        c = (double)method.invoke(cm);
-      } catch (Exception e) {
-        throw new H2OIllegalArgumentException(
-                "Failed to get metric: " + criterion + " from ModelMetrics object: " + mm,
-                "Failed to get metric: " + criterion + " from ModelMetrics object: " + mm + ", criterion: " + method + ", exception: " + e
-        );
-      }
+      c = (double) method.invoke(obj);
```
[the remark this answers is not in the document]
Comment (michalkurka) [reply h2o-core/src/main/java/hex/ModelMetrics.java]: fixed. thank you

File: h2o-core/src/main/java/hex/ModelMetrics.java [hunk 3]
```diff
@@ -127,19 +127,25 @@ static public double getMetricFromModel(Key<Model> key, String criterion) {
   }
 
   static public double getMetricFromModelMetric(ModelMetrics mm, String criterion) {
-    if (null == criterion || criterion.equals("")) throw new H2OIllegalArgumentException("Need a valid criterion, but got '" + criterion + "'.");
+    if (null == criterion || criterion.equals(""))
+      throw new H2OIllegalArgumentException("Need a valid criterion, but got '" + criterion + "'.");
+
```
Comment (pkozelka) [inline h2o-core/src/main/java/hex/ModelMetrics.java]: this hunk seems to be reformat-only; wouldn't you consider adding curly braces?
Maybe my personal problem, but I find the python-like formatting slightly confusing in Java code.

File: h2o-core/src/main/java/hex/ModelMetrics.java [hunk 4]
```diff
@@ -127,19 +127,25 @@ static public double getMetricFromModel(Key<Model> key, String criterion) {
...
+    Object obj = null;
+    criterion = criterion.toLowerCase();
     ConfusionMatrix cm = mm.cm();
     try {
-      method = mm.getClass().getMethod(criterion.toLowerCase());
+      method = mm.getClass().getMethod(criterion);
+      obj = mm;
     }
     catch (Exception e) {
       // fall through
```
Comment (pkozelka) [inline h2o-core/src/main/java/hex/ModelMetrics.java]: I am aware that this is outside your PR, just noting: here the exception is _ignored_, perhaps it's my poor English but _fall through_ gives me completely different intuition for handling than what happens here. 
Besides, such frequent exception swallowing, even in obvious cases like this, is a little bit disturbing. I guess that in most these cases, at least `IllegalStateException` could have been thrown.

File: h2o-core/src/main/java/hex/ModelMetrics.java (hunk 3 above)
Comment (michalkurka) [reply h2o-core/src/main/java/hex/ModelMetrics.java]: added

File: h2o-core/src/main/java/hex/ModelMetrics.java (hunk 4 above)
Comment (michalkurka) [reply h2o-core/src/main/java/hex/ModelMetrics.java]: the execution of this method through to the next step ;)

an exception is eventually thrown if neither of the attempts succeeds

**Comment by mjorgen1** [comment]
I am getting this error "Failed to find ModelMetrics for criterion: auc" but for when I run H2OAutoML, after I set sort_metric='auc'. Any solutions for this? @pkozelka kozelka @wendycwong
